yaml/yaml_test.go

```diff
@@ -21,95 +21,77 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 
 	"github.com/elastic/go-ucfg"
 	"github.com/elastic/go-ucfg/cfgtest"
 )
 
 func TestPrimitives(t *testing.T) {
-	input := []byte(`
+	input := `
     b: true
     i: 42
     u: 23
     f: 3.14
     s: string
-  `)
-
-	c, err := NewConfig(input)
-	if err != nil {
-		t.Fatalf("failed to parse input: %v", err)
-	}
-
+  `
+	c := mustNewConfig(t, input)
 	verify := struct {
 		B bool
 		I int
 		U uint
 		F float64
 		S string
 	}{}
-	err = c.Unpack(&verify)
-	assert.Nil(t, err)
+	err := c.Unpack(&verify)
+	require.NoError(t, err, "failed to unpack config")
 
-	assert.Equal(t, true, verify.B)
+	assert.True(t, verify.B)
 	assert.Equal(t, 42, verify.I)
 	assert.Equal(t, uint(23), verify.U)
 	assert.Equal(t, 3.14, verify.F)
 	assert.Equal(t, "string", verify.S)
 }
 
 func TestNested(t *testing.T) {
-	input := []byte(`
+	input := `
     c:
       b: true
-  `)
-
-	c, err := NewConfig(input)
-	if err != nil {
-		t.Fatalf("failed to parse input: %v", err)
-	}
-
+  `
+	c := mustNewConfig(t, input)
 	var verify struct {
 		C struct{ B bool }
 	}
-	err = c.Unpack(&verify)
-	assert.NoError(t, err)
+	err := c.Unpack(&verify)
+	require.NoError(t, err, "failed to unpack config")
 	assert.True(t, verify.C.B)
 }
 
 func TestNestedPath(t *testing.T) {
-	input := []byte(`
+	input := `
     c.b: true
-  `)
-
-	c, err := NewConfig(input, ucfg.PathSep("."))
-	if err != nil {
-		t.Fatalf("failed to parse input: %v", err)
-	}
-
+  `
+	c := mustNewConfig(t, input, ucfg.PathSep("."))
 	var verify struct {
 		C struct{ B bool }
 	}
-	err = c.Unpack(&verify)
-	assert.NoError(t, err)
+	err := c.Unpack(&verify)
+	require.NoError(t, err, "failed to unpack config")
 	assert.True(t, verify.C.B)
 }
 
 func TestArray(t *testing.T) {
-	input := []byte(`
+	input := `
 - b: 2
   c: 3
 - c: 4
-`)
-
-	c, err := NewConfig(input)
-	if err != nil {
-		t.Fatalf("failed to parse input: %v", err)
-	}
-
+`
+	c := mustNewConfig(t, input)
 	var verify []map[string]int
-	err = c.Unpack(&verify)
-	assert.Nil(t, err)
+	err := c.Unpack(&verify)
+	require.NoError(t, err, "failed to unpack config")
```

Comment:
this is a repeating pattern. Makes me wonder if we should also provide a `mustUnpack(t, c, &verify)`.

Reply:
Yeah, that is a fair point and sounds reasonable.